File: Greedy/c++/14_곱하기혹은더하기.cpp
// 이것이 취업을 위한 코딩테스트다 312p
// 2

#include <iostream>

using namespace std;

int main()
{
	string s;
	cin >> s;
	
	long long result = s[0] - '0';
	for (int i = 1; i < s.size(); i++) {
		int num = s[i] - '0';
		if (num <= 1 || result <= 1)
			result += num;
		else
			result *= num;
	}
	cout << result;
}

File: Greedy/python/14_곱하기혹은더하기.py
# 이것이 취업을 위한 코딩테스트다 312p
# 1

S = input()
result = int(S[0])
for i in range(1, len(S)):
    if int(S[i]) <= 1 or result <= 1:
        result += int(S[i])
    else:
        result *= int(S[i])
print(result)